crates/solvers/src/infra/dex/mod.rs

```diff
@@ -20,18 +22,19 @@ impl Dex {
         &self,
         order: &dex::Order,
         slippage: &dex::Slippage,
-        gas_price: auction::GasPrice,
+        auction: &Auction,
     ) -> Result<dex::Swap, Error> {
         let swap = match self {
-            Dex::Balancer(balancer) => balancer.swap(order, slippage, gas_price).await?,
-            Dex::OneInch(oneinch) => oneinch.swap(order, slippage, gas_price).await?,
-            Dex::ZeroEx(zeroex) => zeroex.swap(order, slippage, gas_price).await?,
+            Dex::Balancer(balancer) => balancer.swap(order, slippage, auction).await?,
+            Dex::OneInch(oneinch) => oneinch.swap(order, slippage, auction).await?,
+            Dex::ZeroEx(zeroex) => zeroex.swap(order, slippage, auction).await?,
```

Comment:
I think only paraswap actually needs the additional metadata of the auction to get the decimals. All the other implementations could continue to just take the gas price.

Reply:
Why would they? The `Auction` is available to the DEX `swap` process. I'm keeping all of the interfaces consistent.

Reply:
IMO it makes sense to only pass around what you actually need in a function.
To me it's a deficiency of the paraswap API that you have to specify the decimals with your request which then leaks into the remaining dex implementations.
I understood the `infra` crate to exist just so we can interface with external APIs and for those the concept of an `Auction` is meaningless.

Reply:
I had a similar comment here: https://github.com/cowprotocol/services/pull/1513/files#r1198763872

Maybe, instead of `dex::slippage::Prices` you need a `dex::Auction` domain type that models the additional auction data that DEX-based solvers care about (so the reference prices and token metadata)?

Reply:
https://github.com/cowprotocol/services/pull/1513/commits/aeb6e3bc819f272f013c007647b143c54de351a4